### DISCUSSION TITLE
Physx5 in Linux using Ubuntu snap o3de?

### DISCUSSION THREAD

**Discussion by lucasw** [o3de/o3de#17652] [Q&A] [answered]
I've tried opening a few projects like https://github.com/RobotecAI/ROSCon2023Demo/ using the snap installed version of O3DE (23.10.2), but I get this error during build

```
    The following dependency requirements could not be satisfied:

  The project requires PhysX5
````

Where do I get that from?

**Comment by lucasw** [comment]
I think I got it, I need to revert ROSCon2023Demo back to a version compatible with 2310.2 (which is `"version": "2.2.1"`, the latest o3de source is 4.2.0)- it looks like one version older than this change which updates it to physx5:

> Adjusments to recent changes in development, mainly PhysX 5 split

https://github.com/RobotecAI/ROSCon2023Demo/commit/b30d5407e48082093df3da8cb1e4efde9ba9f2a3

> The PhysX Gem now independently supports both PhysX 4.1 SDK and PhysX 5.1 SDK concurrently. Previously there was only one PhysX Gem, and version 4.1 of the SDK was the default. 

https://github.com/o3de/o3de/commit/92df6b2afb81d476be516e6e67c0d70df6363f0e



https://github.com/RobotecAI/ROSCon2023Demo/commit/b30d5407e48082093df3da8cb1e4efde9ba9f2a3

----

Alternatively I could take the source of the latest o3de and build it and I expect that would have physx5 in it.

**Comment by spham-amzn** [comment, marked as answer]
The snap package will not work in conjunction with the the ROS2 system. It has something to do with the sandbox that the snap package is running under, and it does not pass in the ros2 environment variables. The debian package should work however, as long as you have installed the [pre-requisite](https://www.docs.o3de.org/docs/welcome-guide/requirements/#linux) first. But make sure to wait until this [PR](https://github.com/o3de/o3de/pull/17623) is merged in and the nightly build is available.

**Reply by lucasw** [reply]
Ok I subscribed to that pr and will grab it from https://o3debinaries.org/development/Latest/Linux/o3de_latest.deb a day after it gets merged.

Is this also mostly the answer to https://robotics.stackexchange.com/questions/110258/add-a-lidar-to-o3de-ros2-project ?  (I think that is the first o3de question there)  In that case I was using the release deb, but maybe I need a tutorial and a very simple example project (e.g. a ros lidar and camera pointing at the default shader ball, but as little else as possible).

-> using the latest nightly deb and o3de-extras did eventually get me to a working ros2 lidar, I added an answer to the above stack question with details
